docs/guide/http_cache.md

```diff
@@ -315,165 +263,186 @@ ezpublish:
+   The application calculates the hash _(`HashGenerator`)_ and then sends a response with the hash in a custom header _(`X-User-Hash` in eZ Platform)_.
+4. The caching proxy receives the hash response, copies the hash header to the client’s original request for `/foo` and restarts the modified original request.
+5. If the response to `/foo` should differ per user context, the application sets a `Vary: X-User-Hash` header, which will make Proxy store the variations of this cache varying on the hash value.
 
-3\. Execute the following command:
 
-`curl -I -H "Accept: application/vnd.fos.user-context-hash" http://<your-domain.com>/_fos_user_context_hash`
+The next time a request comes in from the same user, application lookup for the hash (step 3) won't happen anymore as the
+hash lookup itself is cached by the cache proxy itself.
 
-Remember that you have to send this request to the backend, not to Varnish.
 
-You should get a result like this:
+##### User Context Hash caching explained
 
+Example of response sent to reverse proxy from `/_fos_user_context_hash` with [eZ Platform's default config](#Default-options-for-FOSHttpCacheBundle-defined-in-eZ-Platform):
 ```
 HTTP/1.1 200 OK
-Date: Mon, 03 Oct 2016 15:34:08 GMT
-Server: Apache/2.4.18 (Ubuntu)
-X-Powered-By: PHP/7.0.8-0ubuntu0.16.04.2
-X-User-Hash: b1731d46b0e7a375a5b024e950fdb8d49dd25af85a5c7dd5116ad2a18cda82cb
-Cache-Control: max-age=600, public
-Vary: Cookie,Authorization
+X-User-Hash: <hash>
 Content-Type: application/vnd.fos.user-context-hash
+Cache-Control: public, max-age=600
+Vary: Cookie, Authorization
 ```
 
-4\. Restart the Varnish server.
+As you can see, response is set to be cached for 10minutes. It varies on the `Cookie` header in order to be able to cache
+it for the given user. For cache efficiency reasons our default VCL strips any other cookie then session cookies to make
+this work.
+
+It also varies on `Authorization` to cover any possible basic auth headers in case that is used over sessions for some requests.
+
+!!! note "In case you experience problems with stale user hash"
+
+    In case you notice issues with stale hash usage, before you disable this cache make sure login/logout always
+    generates new session IDs and performs a full redirect in order to make sure no requests are being made using stale
+    user context hashes.
 
-##### Known limitations of the user hash generation
+!!! warning "Known limitations of the user context hash"
 
-If you are using URI-based SiteAccesses matching, the default SiteAccess on the domain needs to point to the same repository, because `/_fos_user_context_hash` is not SiteAccess-aware by default (see `ezpublish.default_router.non_siteaccess_aware_routes` parameter). Varnish does not have knowledge about SiteAccesses, so it won't be able to get user content hash if the default SiteAccess relies on URI.  
+    If you are using URI-based SiteAccesses matching, the default SiteAccess on the domain needs to point to the same
```

Comment:
I honestly don't quite get what you mean here. If using URI-based matching you simply must have a `default_siteaccess` setting, right? And that one is required by kernel. Or are you talking about multi database setup? I haven't thought about that before now, but multi-database setup is indeed a challenge with URI-based matching

Reply:
This is about multi database, will multi repo I imply multi db as well.
Suggestions on making it more clear and readable?

Reply:
Header could maybe be `!!! warning "Known limitations of the user context hash with multi repository setup"`

Reply:
Yes, or in the first sentence says something like "If you are using URI-based SiteAccesses matching on a multi-repo installation ( multiple databases ) ..."
Cause as I read it now, the problem is with URI-based matching, while the actually problem is with multi-repos..
If you use multi-repos and only URI-based matching ( only one domain ), user content hash won't work basically....

Reply:
Updated to reflect suggestion++, so marking this thread as resolved. Please have a look when time permits :)